cluster/juju/layers/kubernetes-master/actions/get-kubeconfig

```diff
@@ -0,0 +1,35 @@
+#!/usr/local/sbin/charm-env python3
+import os
+from yaml import safe_load as load
+from charmhelpers.core.hookenv import (
+    action_get,
+    action_set,
+    action_fail,
+    action_name
+)
+from charmhelpers.core.templating import render
+from subprocess import check_output
+
+
+def _kubectl(args):
+    """
+    Executes kubectl with args as arguments
+    """
+    snap_bin = os.path.join(os.sep, "snap", "bin")
+    env = os.environ.copy()
+    env["PATH"] = os.pathsep.join([snap_bin, env["PATH"]])
+    cmd = ["kubectl", "--kubeconfig=/home/ubuntu/config"]
+    cmd.extend(args)
+    return check_output(cmd, env=env)
+
+
+def get_kubeconfig():
+    """
+    Read the kubeconfig on the master and return it as JSON
```

Comment:
@xiwenc Can you remind me why you want JSON instead of yaml? It might be nice to have a parameter so the user can choose which format they want.

Before I realized that you were converting to JSON, I was wondering if this whole action could be replaced with:
```
#!/bin/bash
action-set "kubeconfig" $(cat /home/ubuntu/config)
```

Reply:
The choice for JSON was the assumption that JSON format is being used in the action response. Other than that we don't have a preference.

The bash oneliner looks tempting but we prefer to use Python with more control on format checking and error handling.